File: backend/src/ai/objects/AIDietWeeklyReportResponse.scala
package delivery.ai.objects

final case class DietNutritionItem(
    name: String,
    amount: String,
    assessment: String
)

final case class DietWeeklySummary(
    calorieTotal: String,
    orderCount: Int,
    topCategory: String,
    topMerchant: String
)

final case class AIDietWeeklyReportResponse(
    summary: DietWeeklySummary,
    nutritionAnalysis: List[DietNutritionItem],
    suggestions: List[String],
    weeklyTrend: String,
    generatedAt: String
)

File: frontend/src/objects/ai/apiTypes/AIDietWeeklyReportResponse.ts
export interface DietNutritionItem {
  name: string
  amount: string
  assessment: string
}

export interface DietWeeklySummary {
  calorieTotal: string
  orderCount: number
  topCategory: string
  topMerchant: string
}

export interface AIDietWeeklyReportResponse {
  summary: DietWeeklySummary
  dietAnalysis: string
  nutritionAnalysis: DietNutritionItem[]
  suggestions: string[]
  weeklyTrend: string
  generatedAt: string
}
